src/app/shared/example-viewer/example-viewer.ts

```diff
@@ -67,6 +67,12 @@ export class ExampleViewer {
   constructor(private readonly snackbar: MatSnackBar, private readonly copier: CopierService) {
   }
 
+  ngOnInit() {
+    if (this.file) {
+      this.file = this.generateUrl(this.file);
```

Comment:
I don't really like this since `file` has different semantics before and after `ngOnInit` runs. I would just save it in a different property like `fileUrl`